src/UI/TransactionsToastList/index.tsx

```diff
@@ -33,49 +54,41 @@ function TransactionsToastList({
   const signedTransactionsToRender =
     signedTransactions || signedTransactionsFromStore;
 
-  const generatedClasses = getGeneratedClasses(
-    className,
-    shouldRenderDefaultCss,
-    {
-      wrapper:
-        'toast-messages d-flex flex-column align-items-center justify-content-sm-end',
-      toast: ''
-    }
-  );
+  const handleDeleteCustomToast = (toastId: string) => {
+    removeToast(toastId);
+    setToastsIds((toastsIds: ToastsType[]): ToastsType[] =>
+      toastsIds.filter((toast: ToastsType) => toast.toastId !== toastId)
+    );
+  };
 
-  const mappedToastsList = toastsIds?.map((toastId: string) => {
-    const currentTx: SignedTransactionsBodyType =
-      signedTransactionsToRender[toastId];
-    if (
-      currentTx == null ||
-      currentTx?.transactions == null ||
-      currentTx?.status == null
-    ) {
-      return null;
-    }
+  const mappedToastsList = toastsIds?.map((props: ToastsType) => {
+    const { toastId, type = ToastsEnum.transaction, message, duration } = props;
+    const args = {
+      [ToastsEnum.custom]: {
+        message: message ?? '',
+        duration,
+        onDelete: () => handleDeleteCustomToast(toastId)
+      },
+      [ToastsEnum.transaction]: {
+        toastId,
+        signedTransactionsToRender,
+        lifetimeAfterSuccess: successfulToastLifetime
+      }
+    }[type as string];
```

Comment:
This logic is a bit convulted. Maybe we should just do a switch and return <Toast with different props?